src/Compiler/ReleaseNotesBuilder.cs

```diff
@@ -26,42 +20,44 @@ public ReleaseNotesBuilder(IGitHubClient gitHubClient, string user, string repos
             this.milestoneTitle = milestoneTitle;
         }
 
-        public static string LabelPrefix
-        {
-            get { return "Type: "; }
-        }
+        public static string LabelPrefix => "Type: ";
 
         public async Task<string> BuildReleaseNotes()
         {
-            LoadMilestones();
+            var milestones = await gitHubClient.GetMilestones();
+
+            var targetMilestone = milestones.FirstOrDefault(x => x.Title == milestoneTitle);
 
-            GetTargetMilestone();
-            var issues = await GetIssues(targetMilestone);
+            if (targetMilestone == null)
+            {
+                throw new Exception($"Could not find milestone for '{milestoneTitle}'.");
+            }
+            var issues = await gitHubClient.GetIssues(targetMilestone);
             var stringBuilder = new StringBuilder();
-            var previousMilestone = GetPreviousMilestone();
+            var previousMilestone = GetPreviousMilestone(targetMilestone, milestones);
             var numberOfCommits = await gitHubClient.GetNumberOfCommitsBetween(previousMilestone, targetMilestone);
 
             if (issues.Count > 0)
             {
-                var issuesText = String.Format(issues.Count == 1 ? "{0} issue" : "{0} issues", issues.Count);
+                var issuesText = Format(issues.Count == 1 ? "{0} issue" : "{0} issues", issues.Count);
 
                 if (numberOfCommits > 0)
                 {
-                    var commitsLink = GetCommitsLink(previousMilestone);
-                    var commitsText = String.Format(numberOfCommits == 1 ? "{0} commit" : "{0} commits", numberOfCommits);
+                    var commitsLink = GetCommitsLink(targetMilestone, previousMilestone);
+                    var commitsText = Format(numberOfCommits == 1 ? "{0} commit" : "{0} commits", numberOfCommits);
 
-                    stringBuilder.AppendFormat(@"As part of this release we had [{0}]({1}) which resulted in [{2}]({3}) being closed.", commitsText, commitsLink, issuesText, targetMilestone.HtmlUrl());
+                    stringBuilder.Append($"As part of this release we had [{commitsText}]({commitsLink}) which resulted in [{issuesText}]({targetMilestone.HtmlUrl()}) being closed.");
```

Comment:
tbh I'm not a big fan of the commit numbers. Even for a single bugfix it results in at least 2 commits because of the additional merge commit. How about removing that part?